os/se/sss/security_hal_wrapper.c

```diff
@@ -0,0 +1,1637 @@
+		break;
+	case HAL_ECDSA_BRAINPOOL_P384R1:
+	case HAL_ECDSA_BRAINPOOL_P512R1:
+		return HAL_NOT_SUPPORTED;
+	case HAL_ECDSA_SEC_P192R1:
+		ecc_pub.curve = OID_ECC_P192;
+		break;
+	case HAL_ECDSA_SEC_P224R1:
+		ecc_pub.curve = OID_ECC_P224;
+		break;
+	case HAL_ECDSA_SEC_P256R1:
+		ecc_pub.curve = OID_ECC_P256;
+		break;
+	case HAL_ECDSA_SEC_P384R1:
+		ecc_pub.curve = OID_ECC_P384;
+		break;
+	case HAL_ECDSA_SEC_P512R1:
+		ecc_pub.curve = OID_ECC_P521;
+		break;
+	case HAL_ECDSA_UNKNOWN:
+		return HAL_INVALID_ARGS;
+	}
+	ecc_pub.publickey_x = ecdh_param->pubkey_x->data;
+	ecc_pub.x_byte_len = ecdh_param->pubkey_x->data_len;
+	ecc_pub.publickey_y = ecdh_param->pubkey_y->data;
+	ecc_pub.y_byte_len = ecdh_param->pubkey_y->data_len;
+
+	unsigned char output[HAL_MAX_SHARED_SECRET_LEN];
+
+	ISP_CHECKBUSY();
+	ret = isp_compute_ecdh_securekey(output, &shared_secret->data_len, ecc_pub, key_idx);
+	if (ret != 0) {
+		isp_clear(0);
+		HWRAP_LOG("ISP failed (%zu)\n", ret);
+		return HAL_FAIL;
+	}
+
+	HAL_COPY_DATA(shared_secret, output, shared_secret->data_len);
+
+	return HAL_SUCCESS;
+}
+
+int sss_hal_set_certificate(uint32_t cert_idx, hal_data *cert_in)
+{
+	HWRAP_ENTER;
+	uint32_t ret;
+	unsigned char *cert = cert_in->data;
+	unsigned int cert_len = cert_in->data_len;
+
+	ISP_CHECKBUSY();
+	ret = isp_write_cert(cert, cert_len, cert_idx);
+	if (ret != 0) {
+		isp_clear(0);
+		HWRAP_LOG("ISP failed (%zu)\n", ret);
+		return HAL_FAIL;
+	}
+
+	return HAL_SUCCESS;
+}
+
+int sss_hal_get_certificate(uint32_t cert_idx, hal_data *cert_out)
+{
+	HWRAP_ENTER;
+	uint32_t ret;
+	unsigned char buf[HAL_MAX_BUF_SIZE];
+	unsigned int buf_len;
+
+	ISP_CHECKBUSY();
+	if (cert_idx == FACTORYKEY_ARTIK_CERT) {
+		ret = isp_get_factorykey_data(buf, &buf_len, cert_idx);
+		if (ret != 0) {
+			isp_clear(0);
+			HWRAP_LOG("ISP failed (%zu)\n", ret);
+			return HAL_FAIL;
+		}
+	} else {
+		ret = isp_read_cert(buf, &buf_len, cert_idx);
+		if (ret != 0) {
+			isp_clear(0);
+			HWRAP_LOG("ISP failed (%zu)\n", ret);
+			return HAL_FAIL;
+		}
+	}
+
+	HAL_COPY_DATA(cert_out, buf, buf_len);
+
+	return HAL_SUCCESS;
+}
+
+int sss_hal_remove_certificate(uint32_t cert_idx)
+{
+	HWRAP_ENTER;
+	return HAL_NOT_SUPPORTED;
+}
+
+int sss_hal_get_factory_key(uint32_t key_idx, hal_data *key)
+{
+	HWRAP_ENTER;
+	uint32_t ret;
+	if (key_idx == FACTORYKEY_ARTIK_DEVICE) {
+		ISP_CHECKBUSY();
+		ret = isp_get_factorykey_data(key->data, &key->data_len, key_idx);
+		if (ret != 0) {
+			isp_clear(0);
+			HWRAP_LOG("ISP failed (%zu)\n", ret);
+			return HAL_FAIL;
+		}
+	} else {
+		return HAL_NOT_SUPPORTED;
+	}
+
+	return HAL_SUCCESS;
+}
+
+int sss_hal_get_factory_cert(uint32_t cert_idx, hal_data *cert)
+{
+	HWRAP_ENTER;
+	uint32_t ret;
+	if (cert_idx == FACTORYKEY_ARTIK_CERT) {
+		ISP_CHECKBUSY();
+		ret = isp_get_factorykey_data(cert->data, &cert->data_len, cert_idx);
+		if (ret != 0) {
+			isp_clear(0);
+			HWRAP_LOG("ISP failed (%zu)\n", ret);
+			return HAL_FAIL;
+		}
+	} else {
```

Comment:
How about below style?
if (A != B) {
   return error;
}
do_something;
if (C != D) {
   return error;
}
return OK;
```

Reply:
it's not our code so it would be better not to modify it.